client/state/happychat/test/middleware.js

```diff
@@ -3,17 +3,33 @@
  */
 import { expect } from 'chai';
 import deepFreeze from 'deep-freeze';
-import { stub } from 'sinon';
+import { spy, stub } from 'sinon';
 
 /**
  * Internal dependencies
  */
 import {
+	HAPPYCHAT_SET_MESSAGE,
 	HAPPYCHAT_TRANSCRIPT_RECEIVE,
 } from 'state/action-types';
-import { requestTranscript } from '../middleware';
+import middleware, { requestTranscript } from '../middleware';
 
 describe( 'middleware', () => {
+	describe( 'HAPPYCHAT_SET_MESSAGE action', () => {
+		it( 'should send the connection a typing signal when a message is present', () => {
+			const action = { type: HAPPYCHAT_SET_MESSAGE, message: 'Hello world' };
+			const connection = { typing: spy() };
+			middleware( connection )()( spy() )( action );
```

Comment:
Since we are not really interested in what's happening to the `next` function here, could we consider to use `noop` from `lodash` instead of creating a `spy` instance here? It would be a little bit light weighted and the "don't care" intention would be clearer.